src/tests/playwright/pages/util/recorder/recorder-main.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview AudioWorklet Recorder Node to record raw PCM data from
+ * Web Audio Graph.
+ */
+
+/**
+ * Creates an AudioWorklet recorder node to record input for a specified
+ * length of time (seconds). Passes audio through to output.
+ * @param {AudioContext} context - The AudioContext to create the recorder node.
+ * @param {number} recordLength - The length in seconds to record for.
+ * @return {Object} An object containing:
+ *   - {AudioWorkletNode} recorder: The recorder AudioWorkletNode.
+ *   - {Promise<AudioBuffer>} bufferPromise: A promise that resolves to an
+ *   AudioBuffer once filled.
+ */
+export const record = async (context, recordLength) => {
+  console.assert(context instanceof AudioContext);
+  console.assert(typeof recordLength === 'number' && recordLength > 0);
+
+  const maxSamples = recordLength * context.sampleRate;
+
+  await context.audioWorklet.addModule('./util/recorder/recorder-processor.js');
+
+  const recorder = new AudioWorkletNode(context, 'recorder', {
+    processorOptions: {
+      maxSamples,
+    },
+  });
+
+  let bufferResolve;
+  recorder.port.onmessage = (e) => {
+    if (e.data.message === 'RECORD_DONE') {
+      const channelBuffers = e.data.buffer;
+      const audioBuffer = new AudioBuffer({
```

Comment:
What's this `audioBuffer` for? Can we rename this for the purpose of the variable?

It looks like a transitory buffer to store the rendered result. Right?

Reply:
This `audioBuffer` is returned by the recorder back to the user.

Reply:
Can we use Float32Array instead of AudioBuffer? What's the benefit of using an AudioBuffer object?

Reply:
Since we are recording multiple channels of audio, I thought it would be cleaner to wrap everying into an AudioBuffer. The alternative was returning `Array<Float32Array>`

Reply:
SGTM!

Reply:
Getting Float32Array back from AudioBuffer involves with `getChannelData()`. So my preference is to use `Array<Float32Array>` for easier handling afterwards.